### PR TITLE
[Paywalls V2] Allowing intro eligibility overrides for text (and image and stack)

### PR DESCRIPTION
**Pull request by joshdholtz**
### Motivation

Need to show different text when intro offer is eligible

### Description

New `IntroEligibilityContext` object that will store the eligibility cache for the paywall components to reference

- Reused `TrialOrIntroEligibilityChecker` from Paywalls V1
- Stored as a `StateObject` on the paywall parent
- Passed as an environment object to children components
- Eligibility is fetched on load (async) and will be propagated to children components
    - All eligibility should already be cached in the SDK so it should be fairly instant

### CODE REVIEW

File: RevenueCatUI/Templates/Components/PresentedPartials.swift [hunk 1]
```diff
@@ -72,12 +72,23 @@ extension PresentedPartial {
...
         with presentedOverrides: PresentedOverrides<Self>?
     ) -> Self? {
-        let conditionPartial = buildConditionPartial(for: condition, with: presentedOverrides)
+        var conditionPartial = buildConditionPartial(for: condition, with: presentedOverrides)
 
-        guard case .selected = state else { return conditionPartial }
-        return Self.combine(conditionPartial, with: presentedOverrides?.states?.selected)
+        if isEligibleForIntroOffer {
+            conditionPartial = Self.combine(conditionPartial, with: presentedOverrides?.introOffer)
+        }
```
Comment (joshdholtz) [inline RevenueCatUI/Templates/Components/PresentedPartials.swift:82]: This is the core part of this change.

The `introOffer` partial will be applied **after** the screen class and **before** the selected state if an intro offer is available.

Any of the components that use partials (text, image, and stack) will gain this ability.

File: RevenueCatUI/Templates/Components/IntroOfferEligibilityContext.swift [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    private(set) var allEligibility: [Package: IntroEligibilityStatus] = [:]
+
+    init(introEligibilityChecker: TrialOrIntroEligibilityChecker) {
+        self.introEligibilityChecker = introEligibilityChecker
+    }
+
+    func computeEligibility(for packages: [Package]) async {
+        let result = await self.introEligibilityChecker.eligibility(for: packages)
+
+        print("JOSH \(result)")
```
[the remark this answers is not in the document]
Comment (JayShortway) [reply RevenueCatUI/Templates/Components/IntroOfferEligibilityContext.swift:34]: 😿

File: RevenueCatUI/Templates/Components/IntroOfferEligibilityContext.swift [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+#if PAYWALL_COMPONENTS
+
+@MainActor
+class IntroOfferEligibilityContext: ObservableObject {
+
+    private let introEligibilityChecker: TrialOrIntroEligibilityChecker
+
+    @Published
+    private(set) var allEligibility: [Package: IntroEligibilityStatus] = [:]
```
Comment (JayShortway) [inline RevenueCatUI/Templates/Components/IntroOfferEligibilityContext.swift:25]: How about `allEligibilities`, or just `all`?

File: RevenueCatUI/Templates/Components/PackageValidator.swift [hunk 4]
```diff
@@ -11,6 +11,7 @@
 //
 //  Created by Josh Holtz on 10/25/24.
 
+import Combine
```
Comment (JayShortway) [inline RevenueCatUI/Templates/Components/PackageValidator.swift:14]: Is this needed? Seem to be no other changes in this file other than this import.

**Review by JayShortway — COMMENTED** [review]
Nice nice! One naming suggestion and an import to double check.

**Review by JayShortway — APPROVED** [review]
Looking good!

**Pull request merged**